filip/models/ngsi_v2/registrations.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+"""
+Created on Sep 15, 2021
+
+@author Thomas Storek
+
+This module contains NGSIv2 models for context registrations in the context
+broker.
+"""
+from typing import List, Union, Optional
+from datetime import datetime
+from aenum import Enum
+from pydantic import \
+    AnyHttpUrl, \
+    BaseModel, \
+    Field
+from filip.models.ngsi_v2.context import EntityPattern, Expression
+from filip.models.ngsi_v2.subscriptions import Status
+
+
+class ForwardingMode(str, Enum):
+    _init_ = 'value __doc__'
+
+    NONE = "none", "This provider does not support request forwarding."
+    QUERY = "query", "This provider only supports request forwarding to " \
+                     "query data."
+    UPDATE = "update", "This provider only supports request forwarding to " \
+                       "update data."
+    ALL = "all", "This provider supports both query and update forwarding " \
+                 "requests. (Default value)"
+
+
+class Provider(BaseModel):
+    http: AnyHttpUrl = Field(
+        description="It is used to convey parameters for providers that "
```

Comment:
In Issue #29 I was currently fixing the issue here.
Provider does need an url field inside the http field containing the given httpadresse. Same as Notification in the subscription class, where the HTTP class was used.

Maybe the HTTP class should be on a more global scale so that the Provider can here also use it.

As the file was drastically changed in location and name, I suggest that it is for the best if we discard the changes made in Issue #29 and you simply add these changes here. A small test for registrations would also be very useful (I just started with it in Issue 29)